sdk/src/network-client.ts

```diff
@@ -1198,24 +1198,59 @@ class AleoNetworkClient {
    * const transaction = await networkClient.waitForTransactionConfirmation(transactionId);
    */
   async waitForTransactionConfirmation(
-      transactionId: string,
-      checkInterval: number = 2000,  // Poll every 2 seconds
-      timeout: number = 45000        // Timeout after 45 seconds
-  ): Promise<Transaction> {
+    transactionId: string,
+    checkInterval: number = 2000,
+    timeout: number = 45000
+  ): Promise<ConfirmedTransactionJSON> {
     const startTime = Date.now();
-
-    return new Promise<Transaction>((resolve, reject) => {
+  
+    return new Promise((resolve, reject) => {
       const interval = setInterval(async () => {
+        const elapsed = Date.now() - startTime;
+  
+        if (elapsed > timeout) {
+          clearInterval(interval);
+          return reject(new Error("Transaction confirmation timed out"));
+        }
+  
         try {
-          // Replace with actual Aleo transaction lookup API
-          const transaction = <Transaction>await this.getTransactionObject(transactionId);
-          resolve(transaction);
-          if (Date.now() - startTime > timeout) {
+          const res = await fetch(`${this.host}/transaction/confirmed/${transactionId}`, {
+            headers: this.headers
+          });
+          if (!res.ok) {
+            let text = "";
+            try {
+              text = await res.text();
+              console.warn("Response text from server:", text);
+            } catch (err) {
+              console.warn("Failed to read response text:", err);
+            }
+          
+            if (res.status >= 400 && res.status < 500 && text.includes("Invalid URL")) {
+              clearInterval(interval);
+              return reject(new Error(`Malformed transaction ID: ${text}`));
+            }
+          
+            // Log and continue polling for 404s or 5XX errors in case a tx doesn't exist yet
+            console.warn("Non-OK response (retrying):", res.status, text);
+            return;
+          }
+          
+          const text = await res.text();
+          const data = parseJSON(text);          
+          if (data?.status === "accepted") {
+            clearInterval(interval);
+            return resolve(data);
+          }
+        
+          if (data?.status === "rejected") {
             clearInterval(interval);
-            reject(new Error("Transaction confirmation timed out"));
+            return reject(new Error(`Transaction ${transactionId} was rejected by the network`));
```

Comment:
Also here we probably want to add more context, the below is just a suggestion, but in general we should explain the things the developer should check for to resolve the failure.

```suggestion
            return reject(new Error(`Transaction ${transactionId} was rejected by the network - ensure the account paying the fee has enough credits and that inputs to the on-chain part of the function are correct`));
```